src/main/java/vg/civcraft/mc/citadel/ReinforcementManager.java

```diff
@@ -2,36 +2,49 @@
 
 import org.bukkit.Location;
 import org.bukkit.block.Block;
-
 import vg.civcraft.mc.citadel.model.CitadelChunkData;
 import vg.civcraft.mc.citadel.model.Reinforcement;
 import vg.civcraft.mc.civmodcore.locations.chunkmeta.api.BlockBasedChunkMetaView;
 import vg.civcraft.mc.civmodcore.locations.chunkmeta.block.table.TableBasedDataObject;
 import vg.civcraft.mc.civmodcore.locations.chunkmeta.block.table.TableStorageEngine;
 
+import java.util.Optional;
+
 public class ReinforcementManager {
 
 	private BlockBasedChunkMetaView<CitadelChunkData, TableBasedDataObject, TableStorageEngine<Reinforcement>> chunkMetaData;
 
-	ReinforcementManager(
-			BlockBasedChunkMetaView<CitadelChunkData, TableBasedDataObject, TableStorageEngine<Reinforcement>> chunkMetaData) {
+	ReinforcementManager(BlockBasedChunkMetaView<CitadelChunkData, TableBasedDataObject, TableStorageEngine<Reinforcement>> chunkMetaData) {
 		this.chunkMetaData = chunkMetaData;
 	}
 
 	/**
 	 * Gets the reinforcement at the given location if one exists
-	 * 
+	 *
 	 * @param location Location to get reinforcement for
 	 * @return Reinforcement at the location or null if no reinforcement exists
-	 *         there
+	 * there
 	 */
 	public Reinforcement getReinforcement(Location location) {
-		return (Reinforcement) chunkMetaData.get(location);
```

Comment:
This introduces new problems while not solving the source of the problem its trying to address.

Gets on the cache should always get whatever is in the cache, if null is returned there should not be a reinforcement at that location. If you return null, you should also delete the reinforcement, otherwise key constraints are gonna fuck you up when creating new reinforcements at that location etc.

Also even if we did delete the reinforcement, it'd be only a cosmetic fix on top of ChunkMetas caching issue, which should be fixed instead as it might manifest elsewhere as well.